packages/core/src/modules/oob/OutOfBandModule.ts

```diff
@@ -0,0 +1,297 @@
+import type { AgentMessage } from '../../agent/AgentMessage'
+import type { AgentMessageReceivedEvent } from '../../agent/Events'
+import type { Logger } from '../../logger'
+import type { PlaintextMessage } from '../../types'
+import type { ConnectionRecord } from '../connections'
+
+import { parseUrl } from 'query-string'
+import { EmptyError } from 'rxjs'
+import { Lifecycle, scoped } from 'tsyringe'
+
+import { Dispatcher } from '../../agent/Dispatcher'
+import { EventEmitter } from '../../agent/EventEmitter'
+import { AgentEventTypes } from '../../agent/Events'
+import { MessageSender } from '../../agent/MessageSender'
+import { createOutboundMessage } from '../../agent/helpers'
+import { AriesFrameworkError } from '../../error'
+import { ConsoleLogger, LogLevel } from '../../logger'
+import { ConnectionInvitationMessage, DidCommService, ConnectionState, ConnectionsModule } from '../connections'
+import { MediationRecipientService } from '../routing'
+
+import { HandshakeReuseHandler } from './handlers'
+import { OutOfBandMessage, HandshakeReuseMessage } from './messages'
+
+interface CreateOutOfBandMessageConfig {
+  label?: string
+  goalCode?: string
+  goal?: string
+  handshake: boolean
+  multiUseInvitation: boolean
+}
+
+interface ReceiveOutOfBandMessageConfig {
+  autoAccept: boolean
```

Comment:
I was thinking whether we should also call this `autoAcceptConnection`? If we don't set `autoAccept`: true we currently still process the messages. Whether they are auto accepted depends on the agent config (`autoAcceptCredentials`, `autoAcceptProofs`, etc...). `autoAccept` makes me think  the `autoAccept` property has influence on the accepance of the `requests~attach` messages

Reply:
Yeah, I renamed it to `autoAcceptConnection`. Would it make sense to eventually have a similar `autoAccept` config for `requests~attach` messages?